### PR TITLE
scaleBar enhancements

### PR DESCRIPTION
**Pull request by nnhubbard**
This includes enhancements to the `scaleBar` that allow updating labels colors to better work with dark maps, and also allowing the user to control if the `scaleBar` is using metric.

## Dark Mode Enhancement

Light map with white outlined labels and black text:

mi

Dark map with black outlined labels and white text:

dark

`MGLMapView` includes a new property, `scaleBarShouldShowDarkStyles` to update the `scaleBar`.

## Metric Enhancement

Imperial units:

mi

Metric units:

km

`MGLMapView` includes a new property, `scaleBarUsesMetricSystem` to update the `scaleBar`.

### CODE REVIEW

File: platform/ios/platform/ios/src/MGLScaleBar.mm [hunk 1]
```diff
@@ -282,8 +301,44 @@ - (MGLRow)preferredRow {
...
 #pragma mark - Setters
 
+- (void)setUsesMetricSystem:(BOOL)usesMetricSystem {
+    
+    if (_usesMetricSystem != usesMetricSystem) {
+        
+        _usesMetricSystem = usesMetricSystem;
+        
+        // Set the distance formatter locale using Germany for metric and United States for imperial.
+        self.formatter.numberFormatter.locale = usesMetricSystem ? [NSLocale localeWithLocaleIdentifier:@"de_DE"] : [NSLocale localeWithLocaleIdentifier:@"en_US"];
```
Comment (roblabs) [inline platform/ios/platform/ios/src/MGLScaleBar.mm:330]: This produces the case of `1,5 km`, but is there locale, say in en-US, where I want metric, but a `.` for a decimal point, such as `1.5 km`?

File: platform/ios/platform/ios/src/MGLScaleBar.mm [hunk 2]
```diff
@@ -111,11 +122,19 @@ - (void)drawTextInRect:(CGRect)rect {
     CGContextSetLineJoin(context, kCGLineJoinRound);
     
     CGContextSetTextDrawingMode(context, kCGTextStroke);
-    self.textColor = [UIColor whiteColor];
+    if (_shouldShowDarkStyles) {
+        self.textColor = [UIColor blackColor];
+    } else {
+        self.textColor = [UIColor whiteColor];
+    }
```
Comment (roblabs) [inline platform/ios/platform/ios/src/MGLScaleBar.mm:129]: Great idea to use black and white for appropriate contrasts!!

**Comment by roblabs** [comment]
Approved with the caveat that we may have other locales that need testing.  Great contribution, @nnhubbard.  Thanks for the screen shots!

File: platform/ios/platform/ios/src/MGLScaleBar.mm (hunk 1 above)
Comment (nnhubbard) [reply platform/ios/platform/ios/src/MGLScaleBar.mm:330]: You made a good point about this. I somehow didn't catch this. Will put in a PR to fix this.

**Pull request merged**